deblur/workflow.py

```diff
@@ -401,12 +396,101 @@ def merge_otu_tables(output_fp, all_tables):
     all_tables: list
         list of filepaths for BIOM tables to merge
     """
+    logger = logging.getLogger(__name__)
+    logger.debug('merge_otu_tables for file %d tables'
+                 ' into file %s' % (len(all_tables), output_fp))
+
     master = load_table(all_tables[0])
     for input_fp in all_tables[1:]:
         master = master.merge(load_table(input_fp))
     write_biom_table(master, output_fp)
 
 
+def get_files_for_table(input_dir,
+                        file_end='.fasta.trim.derep.no_artifacts'
+                        '.msa.deblur.no_chimeras'):
+    """Get a list of files to add to the output table
+
+    Parameters:
+    -----------
+    input_dir : string
+        name of the directory containing the deblurred fasta files
+    file_end : string
+        the ending of all the fasta files to be added to the table
+        (default '.trim.derep.no_artifacts.msa.deblur.no_chimeras')
+
+    Returns
+    -------
+    names : list of tuples of (string,string)
+        list of tuples of:
+            name of fasta files to be added to the biom table
+            sampleid (file names without the file_end and path)
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('get_files_for_table input dir %s, '
+                 'file-ending %s' % (input_dir, file_end))
+
+    names = []
+    for cfile in listdir(input_dir):
+        cname = join(input_dir, cfile)
+        if not isfile(cname):
+            continue
+        if len(cfile) < len(file_end):
+            continue
+        if cfile[-len(file_end):] == file_end:
+            sampleid = cfile[:-len(file_end)]
+            names.append((cname, sampleid))
+    logger.debug('found %d files' % len(names))
+    return names
+
+
+def create_otu_table(output_fp, deblurred_list):
+    """Create a biom table out of all files in a directory
+
+    Parameters
+    ----------
+    output_fp : string
+        filepath to final BIOM table
+    deblurred_list : list of string
+        list of file names (including path) of all deblurred
+        fasta files to add to the table
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('create_otu_table for %d samples, '
+                 'into output table %s' % (len(deblurred_list), output_fp))
+
+    seqdict = {}
+    seqlist = []
+    sampdict = {}
+    samplist = []
+    obs = scipy.sparse.dok_matrix((1E9, 1E6), dtype=np.int)
+    for (cfilename, csampleid) in deblurred_list:
+        if csampleid in sampdict:
+            logger.error('sample %s already in table!' % csampleid)
+            continue
+        sampdict[csampleid] = len(sampdict)-1
+        samplist.append(csampleid)
+        csampidx = len(sampdict)-1
+        for chead, cseq in parse_fasta(open(cfilename, 'U')):
+            if cseq not in seqdict:
+                seqdict[cseq] = len(seqlist)
+                seqlist.append(cseq)
+            cseqidx = seqdict[cseq]
+            cfreq = float(re.search('(?<=size=)\w+', chead).group(0))
```

Comment:
I think using the regex is ok - however, it may be useful to first compile the regex outside the for loop (see re.compile) and then just use match/search in here.